plugins/queue/scheduler.js

```diff
@@ -411,48 +422,75 @@ async function start(executor, config) {
             throw value.error;
         }
 
-        const token = executor.tokenGen(Object.assign(tokenConfig, { scope: ['temporal'] }), TEMPORAL_TOKEN_TIMEOUT);
-
-        // set the start time in the queue
-        Object.assign(config, { token });
-        // Store the config in redis
-        await executor.redisBreaker.runCommand('hset', executor.buildConfigTable, buildId, JSON.stringify(config));
-
-        const blockedBySameJob = reach(config, 'annotations>screwdriver.cd/blockedBySameJob', {
-            separator: '>',
-            default: true
-        });
-        const blockedBySameJobWaitTime = reach(config, 'annotations>screwdriver.cd/blockedBySameJobWaitTime', {
-            separator: '>',
-            default: BLOCKED_BY_SAME_JOB_WAIT_TIME
-        });
+        if (isVirtualJob(annotations)) {
+            // Bypass execution of the build if the job is virtual
+            const payload = {
+                status: 'SUCCESS',
+                statusMessage: 'Skipped execution of the virtual job'
+            };
 
-        // Note: arguments to enqueue are [queue name, job name, array of args]
-        enq = await executor.queueBreaker.runCommand('enqueue', executor.buildQueue, 'start', [
-            {
-                buildId,
-                jobId,
-                blockedBy: blockedBy.toString(),
-                blockedBySameJob,
-                blockedBySameJobWaitTime
-            }
-        ]);
-        if (buildStats) {
             await helper
                 .updateBuild(
                     {
                         buildId,
                         token: buildToken,
                         apiUri,
-                        payload: { stats: build.stats, status: 'QUEUED' }
+                        payload
                     },
                     helper.requestRetryStrategy
                 )
                 .catch(err => {
-                    logger.error(`Failed to update build status for build ${buildId}: ${err}`);
+                    logger.error(`virtualBuilds: failed to update build status for build ${buildId}: ${err}`);
 
                     throw err;
                 });
```

Comment:
Can we abstract this section out, the same `helper.updateBuild` is called in either conditions just with different `payload`. Prefer this to be cleaner. 

```
let payload;

if (isVirtualJob(annotations)) {
    // Bypass execution of the build if the job is virtual
    payload = {
        status: 'SUCCESS',
        statusMessage: 'Skipped execution of the virtual job'
    };
} else {
    const token = executor.tokenGen(
        Object.assign(tokenConfig, { scope: ['temporal'] }),
        TEMPORAL_TOKEN_TIMEOUT
    );

    // set the start time in the queue
    Object.assign(config, { token });
    // Store the config in redis
    await executor.redisBreaker.runCommand('hset', executor.buildConfigTable, buildId, JSON.stringify(config));

    const blockedBySameJob = reach(config, 'annotations>screwdriver.cd/blockedBySameJob', {
        separator: '>',
        default: true
    });
    const blockedBySameJobWaitTime = reach(config, 'annotations>screwdriver.cd/blockedBySameJobWaitTime', {
        separator: '>',
        default: BLOCKED_BY_SAME_JOB_WAIT_TIME
    });

    // Note: arguments to enqueue are [queue name, job name, array of args]
    enq = await executor.queueBreaker.runCommand('enqueue', executor.buildQueue, 'start', [
        {
            buildId,
            jobId,
            blockedBy: blockedBy.toString(),
            blockedBySameJob,
            blockedBySameJobWaitTime
        }
    ]);

    if (buildStats) {
        payload = { stats: build.stats, status: 'QUEUED' };
    }
}

await helper
    .updateBuild(
        {
            buildId,
            token: buildToken,
            apiUri,
            payload
        },
        helper.requestRetryStrategy
    )
    .catch(err => {
        if (isVirtualJob(annotations)) {
            logger.error(`virtualBuilds: failed to update build status for build ${buildId}: ${err}`);
        } else {
            logger.error(`Failed to update build status for build ${buildId}: ${err}`);
        }
        throw err;
    });
```

Reply:
Done